### PR TITLE
[Margin] Add MarginExchange class

### PR DESCRIPTION
**Pull request by Herklos**
MarginExchange architecture

Depends on https://github.com/Drakkar-Software/OctoBot-Trading/pull/104

### CODE REVIEW

File: octobot_trading/exchanges/margin/margin_exchange.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+from octobot_trading.exchanges.rest_exchange import RestExchange
+
+
+class MarginExchange(RestExchange):
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/margin/margin_exchange.py]: 👍

File: octobot_trading/exchanges/rest_exchange.py [hunk 2]
```diff
@@ -38,6 +38,8 @@ class RestExchange(AbstractExchange):
     CCXT library wrapper
     """
 
+    CCXT_CLIENT_LOGIN_OPTIONS = {}
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/rest_exchange.py]: What is it used for ? Should we put it in config.json ?

File: octobot_trading/exchanges/rest_exchange.py [hunk 3]
```diff
@@ -276,44 +279,6 @@ def _ensure_order_details_completeness(order, order_required_fields=None):
...
-    # async def get_closed_positions(self, symbol=None, since=None, limit=None, params={}):
-    #     return await self.client.get_closed_positions(symbol=symbol, since=since, limit=limit, params=params)
-    #
-    # async def get_position_trades(self, position_id, symbol=None, since=None, limit=None, params={}):
-    #     return await self.client.get_position_trades(id=position_id, symbol=symbol, since=since, limit=limit,
-    #                                                    params=params)
-
-    async def get_position_status(self, position_id, symbol=None, params={}):
-        return await self.client.get_position_status(id=position_id, symbol=symbol, params=params)
-
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/rest_exchange.py]: 💯

File: octobot_trading/exchanges/rest_exchange.py [hunk 4]
```diff
@@ -381,6 +346,9 @@ def get_exchange_pair(self, pair: str) -> str:
         else:
             raise ValueError(f'{pair} is not supported')
 
+    def set_sandbox_mode(self, is_sandboxed):
+        self.client.setSandboxMode(is_sandboxed)
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/rest_exchange.py]: 👍

File: octobot_trading/exchanges/rest_exchange.py (hunk 2 above)
Comment (Herklos) [reply octobot_trading/exchanges/rest_exchange.py]: It's overwritten by exchange client that need login options like https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/137/files#diff-f9edc36c5fc59e1e37cf878054baf95bR40

**Pull request merged**